internal/integrations/utils.go

```diff
@@ -0,0 +1,194 @@
+package integrations
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+)
+
+const (
+	metricArg      = "metrics"
+	inventoryArg   = "inventory"
+	eventsArg      = "events"
+	exeSuffix      = ".exe"
+	prefixArg      = "--"
+	prefixArgShort = "-"
+)
+
+// MigrateV3toV4Result represents the result of a migration
+type MigrateV3toV4Result struct {
+	MigrateV3toV4Result string `json:"migrateV3toV4Result"`
+}
+
+func migrateV3toV4(pathConfiguration string, pathDefinition string, pathOutput string) string {
+	// Reading old Definition file
+	v3Definition := Plugin{}
+	v3DefinitionBytes, err := readAndUnmarshallConfig(pathDefinition, &v3Definition)
+	if err != nil {
+		log.Fatal(fmt.Errorf("error reading old config definition: %w", err))
+	}
+
+	// Reading old Configuration file
+	v3Configuration := PluginInstanceWrapper{}
+	v3ConfigurationBytes, err := readAndUnmarshallConfig(pathConfiguration, &v3Configuration)
+	if err != nil {
+		log.Fatal(fmt.Errorf("error reading old config configuration: %w", err))
+	}
+
+	// Populating new config
+	v4config, err := populateV4Config(v3Definition, v3Configuration)
+	if err != nil {
+		log.Fatal(fmt.Errorf("error populating new config: %w", err))
+	}
+
+	// Writing output
+	err = writeOutput(pathOutput, v4config, v3DefinitionBytes, v3ConfigurationBytes)
+	if err != nil {
+		log.Fatal(fmt.Errorf("error writing output: %w", err))
+	}
+
+	return fmt.Sprintf("The config has been migrated and placed in: %s", pathOutput)
+}
+
+func readAndUnmarshallConfig(path string, out interface{}) ([]byte, error) {
+	// Reading old definition file
+	bytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("reading %s, %w", path, err)
+	}
+
+	err = yaml.Unmarshal(bytes, out)
+	if err != nil {
+		return nil, fmt.Errorf("unmashalling %s, %w", path, err)
+	}
+
+	return bytes, nil
+}
+
+func populateV4Config(v3Definition Plugin, v3Configuration PluginInstanceWrapper) (*v4, error) {
+	if v3Configuration.IntegrationName != v3Definition.Name {
+		return nil, fmt.Errorf("IntegrationName != Name: %s!=%s", v3Configuration.IntegrationName, v3Definition.Name)
+	}
+
+	// The field os does not have currently a simple way to be migrated
+	if v3Definition.OS != "" {
+		log.Debugf("The old definitions had a os directive, %s. Usually it is not needed, use `when` field otherwhise", v3Definition.OS)
+	}
+
+	v4Config := &v4{}
+	for commandName, pluginV1Command := range v3Definition.Commands {
+		for _, pluginV1Instance := range v3Configuration.Instances {
+			if commandName == pluginV1Instance.Command {
```

Comment:
Should we log an error on `else`?

Reply:
mmm no, it is very common that in the definition there are many commands (example: all_data, metrics, inventory) and in the configuration there is only one